test_plans/execution_model.asciidoc

```diff
@@ -0,0 +1,122 @@
+:sectnums:
+:xrefstyle: short
+
+= Test plan for execution model
+
+This is a test plan for SYCL command groups and execution order and SYCL application memory model as described in Section 3.7.1 and Section 3.8.1 of the SYCL 2020 specification.
+
+== Testing scope
+
+=== Backend coverage
+
+All the tests described below are performed for any SYCL implementation.
+
+=== Device coverage
+
+All of the tests described below are performed only on the default device that
+is selected on the CTS command line.
+
+== Tests
+
+=== Execution order of three command groups submitted to the same queue
+
+Create 2 buffers for large arrays. Submit 2 commands that have some long loops and then fill these buffers with expected values via accessors in loop.
+Submit command that copy content of these buffers to result buffer.
+
+[source,c++]
+----
+queue.submit([&](handler& cgh) {
+  sycl::accessor<int, 1> acc1(buffer1, cgh);
+  cgh.single_task([=] {
+    // long loop
+    for (int i = SIZE; i >= 0; i--) acc1[i] = i;
+  });
+});
+queue.submit([&](handler& cgh) {
+  sycl::accessor<int, 1> acc2(buffer2, cgh);
+  cgh.single_task([=] {
+    // long loop
+    for (int i = SIZE; i >= 0; i--) acc2[i] = i;
+  });
+});
+queue.submit([&](handler& cgh) {
+  sycl::accessor<int, 1> acc1(buffer1, cgh);
+  sycl::accessor<int, 1> acc2(buffer2, cgh);
+  sycl::accessor<int, 1> res_acc1(res_buffer1, cgh);
+  sycl::accessor<int, 1> res_acc2(res_buffer2, cgh);
+  cgh.parallel_for(sycl::range<1>(SIZE), [=](id<1> index) {
+    res_acc1[index] = acc1[index];
+    res_acc2[index] = acc2[index];
+  });
+});
+----
+
+Check that all elements in result buffer are equal to expected values which means that requisites for third command were satisfied before it started to execute.
```

Comment:
I think this test looks good, but I'm surprised we don't already have a test like this from the SYCL 1.2.1 test suite.

Reply:
I couldn't find such checks in CTS unfortunately.